File: backend/src/main/java/com/pawconnect/backend/dog/dto/DogResponse.java
package com.pawconnect.backend.dog.dto;

import java.time.LocalDate;
import java.util.List;

import com.pawconnect.backend.dog.model.ActivityLevel;
import com.pawconnect.backend.dog.model.Gender;
import com.pawconnect.backend.dog.model.Personality;
import lombok.Data;

@Data
public class DogResponse {

    private Long id;
    private String name;
    private String breed;
    private LocalDate birthdate;
    private String size;
    private Gender gender;
    private Personality personality;
    private ActivityLevel activityLevel;
    private String about;
    private List<String> photoUrls;
    private Long ownerId;
}

File: mobile/lib/src/models/dog_response.dart
class DogResponse {
  final int id;
  final String name;
  final String? breed;
  final String? birthdate;
  final String? size;
  final String? gender;
  final String? personality;
  final String? activityLevel;
  final String? about;
  final List<String> photoUrls;
  final int ownerId;

  DogResponse.fromJson(Map<String, dynamic> json)
      : id = json['id'],
        name = json['name'],
        breed = json['breed'],
        birthdate = json['birthdate'],
        size = json['size'],
        gender = json['gender'],
        personality = json['personality'],
        activityLevel = json['activityLevel'],
        about = json['about'],
        photoUrls = List<String>.from(json['photoUrls'] ?? []),
        ownerId = json['ownerId'];
}
